x-pack/plugins/alerting/server/rules_client/lib/build_kuery_node_filter.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fromKueryExpression, KueryNode } from '@kbn/es-query';
+
+export const buildKueryNodeFilter = (filter?: string | KueryNode | null): KueryNode | null => {
+  let optionsFilter: KueryNode | string | null = filter ?? null;
+  try {
+    if (optionsFilter != null && typeof optionsFilter === 'string') {
+      // FUTURE ENGINEER -> if I can parse it that mean it is a KueryNode or it is a string
```

Comment:
instead of the comment you could create a function with a descriptive name like `isKueryNodeOrString`, up to you